thread/common/omrthread.c

```diff
@@ -1998,11 +1995,16 @@ threadAllocate(omrthread_library_t lib, int globalIsLocked)
 static intptr_t
 threadDestroy(omrthread_t thread, int globalAlreadyLocked)
 {
+#if defined(OMR_OS_WINDOWS)
 	omrthread_library_t lib;
+#endif
 
 	ASSERT(thread);
+
+#if defined(OMR_OS_WINDOWS)
 	lib = thread->library;
 	ASSERT(lib);
+#endif
```

Comment:
The var `lib`  is only "actually" used in windows, however this `ASSERT(lib)` line was enabled on every platform. With this change, the assert will only be on windows debug builds. Should I leave the assert enabled on non-windows debug builds? I can switch the ifdef to:

```c++
#if defined(OMR_OS_WINDOWS) || defined(THREAD_ASSERTS)
```

Reply:
Having to #if def all uses of the ASSERT macro can be quite tedious and error prone.  Is there anyway to make the ASSERT no generate a warning if THREAD_ASSERTS is not defined?

Reply:
If the macro ASSERT can not be fixed then I think the above code should have both #if defs included to be clear

Reply:
Updated to include both ifdefs.